Assets/MRTK/SDK/Experimental/Elastic/Scripts/ElasticsManager.cs

```diff
@@ -0,0 +1,222 @@
+                    hostTransform.rotation = rotationElastic.ComputeIteration(targetTransform.Rotation, Time.deltaTime);
+                }
+
+                if (elasticTypes.HasFlag(TransformFlags.Scale))
+                {
+                    hostTransform.localScale = scaleElastic.ComputeIteration(targetTransform.Scale, Time.deltaTime);
+                }
+
+                return elasticTypes;
+            }
+            else 
+            {
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Initialize elastics system with the given host transform.
+        /// Caches a reference to the host transform to be able to keep updating elastics after manipulation.
+        /// </summary>
+        /// <param name="elasticsTransform">host transform the elastics are applied to.</param>
+        public void InitializeElastics(Transform elasticsTransform)
+        {
+            hostTransform = elasticsTransform;
+            if (elasticTypes.HasFlag(TransformFlags.Move))
+            {
+                translationElastic = new VolumeElasticSystem(hostTransform.position,
+                                                             translationElastic?.GetCurrentVelocity() ?? Vector3.zero,
+                                                             translationElasticExtent,
+                                                             translationElasticConfigurationObject.ElasticProperties);
+            }
+            if (elasticTypes.HasFlag(TransformFlags.Rotate))
+            {
+                rotationElastic = new QuaternionElasticSystem(hostTransform.rotation,
+                                                              rotationElastic?.GetCurrentVelocity() ?? Quaternion.identity,
+                                                              rotationElasticExtent,
+                                                              rotationElasticConfigurationObject.ElasticProperties);
+            }
+            if (elasticTypes.HasFlag(TransformFlags.Scale))
+            {
+                scaleElastic = new VolumeElasticSystem(hostTransform.localScale,
+                                                       scaleElastic?.GetCurrentVelocity() ?? Vector3.zero,
+                                                       scaleElasticExtent,
+                                                       scaleElasticConfigurationObject.ElasticProperties);
+            }
+        }
+
+        #region MonoBehaviour Functions
+        private void Update()
+        {
+            // If the user is not actively interacting with the object,
+            // we let the elastic systems continue simulating, to allow
+            // the object to naturally come to rest.
+            if (EnableElasticsUpdate && hostTransform != null)
+            {
+                if (elasticTypes.HasFlag(TransformFlags.Move) && translationElastic != null && translationElastic.GetCurrentVelocity().magnitude > elasticVelocityThreshold)
```

Comment:
Do you know if sqrMagnitude works here over magnitude (i.e. https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html is faster because it avoids the sqrt() call). I guess with small thresholds maybe you get some weird floating point stuff

Reply:
Also this isn't new to this PR, I know this was already stuff that got checked in.

Reply:
oh sure - i'll use squared magnitude with squared threshold :) - good catch

Reply:
threshold is not exposed outside the class - it's a constant , so we should be fine doing that